### ISSUE TITLE
Email confirmation doesn't get send if the user registers with a social login provider

### ISSUE THREAD

**Issue by javiercn** [aspnet/Identity#1637]
* With a social login provider enabled and email confirmation set to required, go to login.
* Click on the social login provider.
* After successful login with your provider, you'll be prompted to introduce your email to finish login/registration.
* Introduce your email and continue.
* No email got sent.
* Logout.
* Try to log-in with your social login provider.
* You'll be prompted again to introduce your email.
* If you try to introduce the same email as before we will tell you that the email is invalid.
* There is no way to complete the registration as you didn't get a confirmation email.
* There is no way to "forget" the association between the social login provider and your unconfirmed email account in order to retry the process.

**Comment by javiercn** [comment]
https://github.com/aspnet/Identity/issues/1667

**Comment by mkArtakMSFT** [comment]
@kichalla , can you please look into this, thanks!

**Comment by javiercn** [comment]
@kichalla Make sure you chat with @haok about this

**Comment by javiercn** [comment]
This doesn't need to be preview2 I think, btw

**Comment by HaoK** [comment]
This could really be 2.2 as well since this is how the templates have worked for a long time

**Comment by mkArtakMSFT** [comment]
@HaoK, can you please look into this? Thanks!

**Comment by HaoK** [comment]
@blowdart @ajcvickers move to 3.0? seems like this is asking for non trivial template changes

**Comment by HaoK** [comment]
Its been like this for years though, email confirmation is off by default, and there's no social providers enabled by default either, I agree that this could be improved, but this doesn't seem important for 2.2 given that its been like this since before identity core and we haven't had many customer requests for this.  The whole email as username thing is related to this, and probably should be addressed all up at the same time in the future.

**Comment by blowdart** [comment]
Yes, 3.0 because we'd have to cope with providers which don't even give you an email address

**Comment by HaoK** [comment]
yeah the basic issue is, the social registration flow is just setting a username that looks like an email, it has nothing to do with the email flow in actuality (except that user name and emails are conflated)

**Comment by thomaslevesque** [comment]
Suggested workflow:
- if we have an email from the external provider, and the user enters the same email in the registration/association form: mark the email as confirmed (because it's probably already confirmed by the external identity provider) (*)
- otherwise, send a confirmation email

_(*) this might not be true for all providers, but it's true at least for Google or AzureAD. Maybe do this only if the `preferred_username` claim is the email, or if the token has an `email_verified` claim?_
